### PR TITLE
osd 4489 app sre pipeline

### PR DESCRIPTION
**Pull request by amolgautam25**
adding template for app-sre pipeline

### CODE REVIEW

File: deploy/template.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          name: aws-account-shredder
+      template:
+        metadata:
+          labels:
+            name: aws-account-shredder
+        spec:
+          serviceAccountName: aws-account-shredder
+          containers:
+            - name: aws-account-shredder
+              image: quay.io/agautam/aws-account-shredder:latest
```
Comment (BumbleFeng) [inline deploy/template.yaml:32]: image: ${IMAGE}:${IMAGE_TAG}

File: deploy/template.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: aws-account-shredder-template
+
+parameters:
+  - name: IMAGE
```
Comment (BumbleFeng) [inline deploy/template.yaml:7]: add `value: quay.io/agautam/aws-account-shredder`

File: deploy/template.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+objects:
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: aws-account-shredder
+      labels:
+        name: aws-account-shredder
+    spec:
+      replicas: 1
```
Comment (BumbleFeng) [inline deploy/template.yaml:21]: Could you expose the REPLICAS  as a parameter?

File: deploy/deployment.yaml [hunk 4]
```diff
@@ -18,6 +18,11 @@ spec:
       containers:
         - name: aws-account-shredder
           image: quay.io/agautam/aws-account-shredder:latest
+          resources:
+            requestes:
+              memory: "1024Mi"
```
Comment (BumbleFeng) [inline deploy/deployment.yaml:23]: Request for 1024Mi is too much for your service. I suggest starting from 100Mi
Btw, you also need to add this part to the template file

File: deploy/template.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    value: latest
+
+objects:
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: aws-account-shredder
+      labels:
+        name: aws-account-shredder
+    spec:
```
Comment (BumbleFeng) [inline deploy/template.yaml:20]: Add update strategy here. for example https://github.com/app-sre/github-mirror/blob/master/openshift/github-mirror.yaml#L24-L28

File: deploy/template.yaml (hunk 3 above)
Comment (amolgautam25) [reply deploy/template.yaml:21]: I have added REPLICAS in parameter

Comment (amolgautam25) [reply deploy/template.yaml:21]: value : "1" . i have to use double quotes here. If i do not use double quotes , i get an error : 
    
 agautam  agautam-workstation  ../aws-account-shredder  oc process -f deploy/template.yaml 
error: failed to read input object (not a Template?): unable to decode "deploy/template.yaml": v1.Template.Parameters: []v1.Parameter: v1.Parameter.Value: ReadString: expects " or n, but found f, error found in #10 byte of ...|,"value":false}]}|..., bigger context ...|est"},{"name":"REPLICAS","required":true,"value":false}]}|...

it seems it is hitting this issue : https://github.com/kubernetes/kubernetes/issues/82296

File: deploy/deployment.yaml (hunk 4 above)
Comment (amolgautam25) [reply deploy/deployment.yaml:23]: changed to 100Mi, also added this to template.yaml

File: deploy/template.yaml (hunk 5 above)
Comment (amolgautam25) [reply deploy/template.yaml:20]: added an update strategy

File: deploy/template.yaml (hunk 3 above)
Comment (BumbleFeng) [reply deploy/template.yaml]: It is ok to add double quotes

File: deploy/template.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      template:
+        metadata:
+          labels:
+            name: aws-account-shredder
+        spec:
+          serviceAccountName: aws-account-shredder
+          containers:
+            - name: aws-account-shredder
+              image: ${IMAGE}:${IMAGE_TAG}
+              imagepullpolicy: always
```
Comment (BumbleFeng) [inline deploy/template.yaml:43]: Could you change to IfNotPresent？

**Comment by BumbleFeng** [comment]
LGTM, except for imagepullpolicy
Once you change that you can merge it

File: deploy/template.yaml (hunk 6 above)
Comment (fahlmant) [reply deploy/template.yaml:43]: Can I ask why `IfNotPresent`? Our operators all use `always`

Comment (BumbleFeng) [reply deploy/template.yaml]: It is not strict. IfNotPresent will help to save some time for restarting pod. 
I am ok with both

Comment (fahlmant) [reply deploy/template.yaml]: :+1:

**Pull request merged**